### PR TITLE
feat: scancel and squeue

### PR DESCRIPTION
**Pull request by db0**
Adds slurm.scancel and slurm.squeue objects

Starting point was code from: https://github.com/amq92/simple_slurm/issues/18

Tested this on a private slurm server I have access to where I used these two classes independently

### CODE REVIEW

**Comment by db0** [comment]
Is this repo not accepting PRs? I notice new commits but no review or comments about this PR.

**Review by dhohn — COMMENTED** [review]
I have tested the squeue Wrapper and it works well for me. I will try the scancel part and report back.

File: simple_slurm/scancel.py [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+            self.sigmtems[job_id] = datetime.now()
+        elif job_id in self.sigmtems:
+            signal = "--signal=KILL"
+            self.sigmkills[job_id] = datetime.now()
+            logger.warning(f"Failed to SIGTERM {job_id}. Sending SIGKILL")
+        # Just straight up kills the node via slurm
+        elif job_id in self.sigmtems:
+            signal = ""
+            logger.warning(f"Failed to SIGKILL {job_id}. Terminating!")
+        result = subprocess.run(["scancel", signal, job_id],
```
Comment (dhohn) [inline simple_slurm/scancel.py:38]: I got a `TypeError: expected str, bytes or os.PathLike object, not int` when using `job_id` as `int`.

File: simple_slurm/scancel.py [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    sigmtems = {}
+    sigmkills = {}
+    stale_delta = None
+
+    def __init__(self, staledelta = timedelta(minutes=30)):
+        self.stale_delta = staledelta
+
+    def cancel_job(self, job_id):
+        '''Sends a straightforward scancel to a job'''
+        result = subprocess.run(["scancel", job_id],
```
Comment (dhohn) [inline simple_slurm/scancel.py:16]: I got a TypeError: expected str, bytes or os.PathLike object, not int when using job_id as int.

File: simple_slurm/scancel.py [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+    def prune_old_jobs(self):
+        '''Clears out signals information older than self.stale_delta'''
+        for signals in [self.sigmtems, self.sigmkills]:
+            for job_id in signals:
+                if self.sigmtems[job_id] < datetime.now() - self.stale_delta:
+                    del self.sigmtems[job_id]
+
+
+    def cancel_all(self, job_id):
```
Comment (dhohn) [inline simple_slurm/scancel.py:51]: Does this need the `job_id` for canceling all?

File: simple_slurm/scancel.py [hunk 4]
```diff
@@ -0,0 +1,56 @@
+import subprocess
+from loguru import logger
```
Comment (dhohn) [inline simple_slurm/scancel.py:2]: Should be declared as dependency right?

**Comment by dhohn** [comment]
@db0 Do you mind if I go and make these suggested changes on your repo/branch? Then merge it here.

**Comment by db0** [comment]
@dhohn go for it.

File: simple_slurm/scancel.py (hunk 4 above)
Comment (db0) [reply simple_slurm/scancel.py:2]: I personaly like using loguru, but maybe it's not wanted. But I doubt this PR will be merged anyway :-S

**Review by dhohn — APPROVED** [review]
Approve.

**Comment by dhohn** [comment]
@amq92 Please have a look and let us know if you can merge.

**Comment by dhohn** [comment]
Dear @amq92 ,

a kind reminder to have a look :)

**Pull request merged**